src/operator/src/req_convert/insert/fill_impure_default.rs

```diff
@@ -0,0 +1,236 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Util functions to help with fill impure default values columns in request
+
+use std::sync::Arc;
+
+use ahash::{HashMap, HashMapExt, HashSet};
+use datatypes::schema::ColumnSchema;
+use snafu::{OptionExt, ResultExt};
+use store_api::storage::{RegionId, TableId};
+use table::metadata::{TableInfo, TableInfoRef};
+
+use crate::error::{ConvertColumnDefaultConstraintSnafu, Result, UnexpectedSnafu};
+use crate::expr_factory::column_schemas_to_defs;
+use crate::insert::InstantAndNormalInsertRequests;
+
+/// Find all columns that have impure default values
+pub fn find_all_impure_columns(table_info: &TableInfo) -> Vec<ColumnSchema> {
+    let columns = table_info.meta.schema.column_schemas();
+    columns
+        .iter()
+        .filter(|column| column.is_default_impure())
+        .cloned()
+        .collect()
+}
+
+/// Fill impure default values in the request
+pub struct ImpureDefaultFiller {
+    impure_columns: HashMap<String, (api::v1::ColumnSchema, Option<api::v1::Value>)>,
+}
+
+impl ImpureDefaultFiller {
+    pub fn new(table_info: TableInfoRef) -> Result<Self> {
+        let impure_column_list = find_all_impure_columns(&table_info);
+        let pks = table_info.meta.primary_key_indices.clone();
+        let pk_names = pks
+            .iter()
+            .map(|&i| table_info.meta.schema.column_name_by_index(i).to_string())
+            .collect::<Vec<_>>();
+        let mut impure_columns = HashMap::new();
+        for column in impure_column_list {
+            let default_value = column
+                .create_impure_default()
+                .context(ConvertColumnDefaultConstraintSnafu {
+                    column_name: column.name.clone(),
+                })?
+                .expect("impure default value");
+            let grpc_default_value = api::helper::to_proto_value(default_value.clone());
```

Comment:
Curious why clippy did not find that
```suggestion
            let grpc_default_value = api::helper::to_proto_value(default_value);
```